### PR TITLE
replacing env_logger with tracing

### PR DESCRIPTION
**Pull request by Shreya111111**
## Feature or Problem
Replacing `env_logger`  with `tracing` to increase the consistency.
#994

### CODE REVIEW

**Comment by Shreya111111** [comment]
@thomastaylor312 can u plz review it.

File: Cargo.toml [hunk 1]
```diff
@@ -154,7 +154,7 @@ tokio-stream = { version = "0.1", default-features = false }
 tokio-tar = { version = "0.3", default-features = false }
 tokio-util = { version = "0.7", default-features = false }
 toml = { version = "0.7", default-features = false }
-tracing = { version = "0.1", default-features = false }
+tracing = { version = "0.1.40", default-features = false }
```
Comment (rvolosatovs) [inline Cargo.toml:157]:
```suggestion
tracing = { version = "0.1", default-features = false }
```

File: vendor/capability-providers/blobstore-s3/tests/large_data.rs [hunk 2]
```diff
@@ -28,7 +28,7 @@ fn gen_bytes(len: usize) -> (Vec<u8>, u32) {
 #[tokio::test]
 async fn run_all() {
     let opts = TestOptions::default();
-    env_logger::try_init().ok();
+    tracing::try_init().ok();
```
Comment (rvolosatovs) [inline vendor/capability-providers/blobstore-s3/tests/large_data.rs:31]: Unfortunately, this won't work - you should use `tracing-subscriber` crate, like e.g. here https://github.com/wasmCloud/wasmCloud/blob/a457cb033cb25f0c29659224db553021244a4298/tests/wasmbus.rs#L462-L469 except we don't need the `,cranelift_codegen=warn,wasmcloud=trace` part

Please make sure to run `cargo test` from the root of this repository to check that it works

**Comment by Shreya111111** [comment]
> Actually, let's not change anything in `vendor`, because that will complicate subtree merges later. We only need this changed in `crates` for now

@rvolosatovs sure, I will look into it

File: crates/wash-cli/src/bin/wash.rs [hunk 3]
```diff
@@ -220,7 +220,7 @@ enum CliCommand {
 #[tokio::main]
 async fn main() {
     use clap::CommandFactory;
-    env_logger::init();
+    tracing_subscriber::init();
```
Comment (thomastaylor312) [inline crates/wash-cli/src/bin/wash.rs:223]: Just a heads up, the simplest substitution here is `tracing_subscriber::fmt::init();` That will give you basically the same behavior as `env_logger::init()`. We can add more advanced filtering and options later if we wish

**Comment by brooksmtownsend** [comment]
Closed in favor of #1153

**Pull request closed without merging**